integration-tests/api/__tests__/store/cart/cart.js

```diff
@@ -1799,6 +1799,53 @@ describe("/store/carts", () => {
       expect(res.data.cart.payment_authorized_at).not.toBe(null)
       expect(res.data.cart.completed_at).not.toBe(null)
     })
+
+    it("completes cart with a non-customer and for a customer with the same email created later the order doesn't show up", async () => {
```

Comment:
suggestion: completes a cart as a guest should not return the order for an logged in account using the same email.